WordPress/AbstractThemeSniff.php

```diff
@@ -0,0 +1,1352 @@
+		}
+		if ( false !== strpos( $file_content, 'wp_head' ) ) {
+			$sniff_helper['basic_function_calls']['wp_head'] = true;
+		}
+		if ( false !== strpos( $file_content, 'language_attributes' ) ) {
+			$sniff_helper['basic_function_calls']['language_attributes'] = true;
+		}
+		if ( false !== strpos( $file_content, 'charset' ) ) {
+			$sniff_helper['basic_function_calls']['charset'] = true;
+		}
+		if ( false !== strpos( $file_content, 'add_theme_support' ) && false !== strpos( $file_content, 'automatic-feed-links' ) ) {
+			$sniff_helper['basic_function_calls']['automatic_feed_links'] = true;
+		}
+		if ( false !== strpos( $file_content, 'comments_template' ) ) {
+			$sniff_helper['basic_function_calls']['comments_template'] = true;
+		}
+		if ( false !== strpos( $file_content, 'wp_list_comments' ) ) {
+			$sniff_helper['basic_function_calls']['wp_list_comments'] = true;
+		}
+		if ( false !== strpos( $file_content, 'comment_form' ) ) {
+			$sniff_helper['basic_function_calls']['comment_form'] = true;
+		}
+		if ( false !== strpos( $file_content, 'body_class' ) ) {
+			$sniff_helper['basic_function_calls']['body_class'] = true;
+		}
+		if ( false !== strpos( $file_content, 'wp_link_pages' ) ) {
+			$sniff_helper['basic_function_calls']['wp_link_pages'] = true;
+		}
+		if ( false !== strpos( $file_content, 'post_class' ) ) {
+			$sniff_helper['basic_function_calls']['post_class'] = true;
+		}
+	}
+
+	/**
+	 * Check for DOCTYPE declaration.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $file_content contains content of file in a single string.
+	 */
+	public function get_doctype_check( $file_content ) {
+		global $sniff_helper;
+		if ( false !== stripos( $file_content, 'DOCTYPE' ) ) {
+			$sniff_helper['doctype'] = true;
+		}
+	}
+
+	/**
+	 * Check that at the very least the following two files exist: index.php
+	 * and style.css.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $themedir contains the theme directory uri.
+	 */
+	public function get_minimum_file_check( $themedir ) {
+		global $sniff_helper;
+		if ( file_exists( $themedir . '/style.css' ) ) {
+			$sniff_helper['style_file_used'] = true;
+		}
+		if ( file_exists( $themedir . '/index.php' ) ) {
+			$sniff_helper['index_file_used'] = true;
+		}
+	}
+
+	/**
+	 * Check if readme.txt or readme.md exists.
+	 * If used then set a flag to true, to be used later.
+	 *
+	 * @param string $themedir contains the theme directory uri.
+	 */
+	public function get_readme_file_check( $themedir ) {
+		global $sniff_helper;
+		if ( file_exists( $themedir . '/readme.txt' ) || file_exists( $themedir . '/readme.md' ) ) {
+			$sniff_helper['readme_file_used'] = true;
+		}
+	}
+
+	/**
+	 * Check for blacklisted files.
+	 * If found place in $sniff_helper array for later use
+	 *
+	 * @param array $themefiles contains the uri of files for the theme.
+	 */
+	public function blacklist_file_check( $themefiles ) {
```

Comment:
I still feel it's best handled in the Class. In your initial issues document it was in the "Rules that would need an alternative solution...." Why was it put there?